### ISSUE TITLE
Button

### ISSUE THREAD

**Issue by mtthwhggns** [apluslms/a-plus-design-system#1]

**Comment by raphendyr** [comment]
Should we have a button group feature? Button group is an element, where buttons don't have any margin between them.

Aslo, should we have a link button? Specially when adding a link to a button group.

Here is an example of the concept. The first line is how I think it should look with our current styles. The second line contains the same with Bootstrap 3 and third line contains the current A+ interface (no button group).

![Buttons in a button group](https://user-images.githubusercontent.com/2327687/90542059-3b03d780-e18c-11ea-9116-8bd8e5e50a77.gif)

**Comment by mtthwhggns** [comment]
@raphendyr Interesting thought. Do you have a use case where it makes sense to use a button group especially? Do you have any concerns other than width about using the current buttons in that case?

I checked Bootstrap docs and the only specific example I found is as a radio button replacement. Users can choose one of several options. It seems good for toolbars where groups of buttons are related to one another (alignment options, types of list, anything where you can pick one from each group).

My main concern about anything where we reduce the margin is that it's easier to accidentally choose the wrong one. We need to consider how users with difficulty using the mouse or are using a small touch screen might be affected (in a similar way to [WCAG SC 2.5.5](https://www.w3.org/WAI/WCAG21/Understanding/target-size)).

My first reaction to the "link button" is that it sounds like it should be a link. As discussed previously, we need to consistently use links for things which behave like links, and the only things to appear like a button should be those which behave like buttons (for more on this see [WCAG SC 3.2.4](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification)).

**Comment by raphendyr** [comment]
> Do you have a use case where it makes sense to use a button group especially?

No, I don't have any really good use cases where this clearly would be needed. Some of the toolbars _could_ possibly benefit from it, but as stated, some of the accessibility concerns might raise. However, it did look quite nice to me in the above video, though with marginal benefits.

In contrast, MOOC-Jutut uses button groups quite much, as there are groups of mutually exclusive selections. Assuming, we would like to use the design system there too, then we probably should cover the case of groups. Though, that can be done when there is actually someone working on it.

> Do you have any concerns other than width about using the current buttons in that case?

The font color for the `aplus-button--default` is invalid on hover (black font). This comes from `.btn-group > .btn:hover`

Additionally, the `btn-link` doesn't have borders, which I think would be beneficial specially in a group. When it's alone, then I think it's better without borders.

> My main concern about anything where we reduce the margin is that it's easier to accidentally choose the wrong one. We need to consider how users with difficulty using the mouse or are using a small touch screen might be affected (in a similar way to WCAG SC 2.5.5).

This was my concern too. This is also an important topic to research on MOOC-Jutut as the UI of that service has gone through few iterations to make it more compact, which might have created problems on this topic.

> My first reaction to the "link button" is that it sounds like it should be a link. As discussed previously, we need to consistently use links for things which behave like links, and the only things to appear like a button should be those which behave like buttons (for more on this see WCAG SC 3.2.4).

Yes exactly. The hair-splitting comes down to if the UI is a web page or a web application. Applications typically have buttons to change views, while web pages have links to other resources.

Nevertheless, to my understanding, the WCAG SC 3.2.4 requires consistency of behaviour. This begs a question which does feel more natural for the user: a button like link or a normal link. In most cases, I would say normal link is better. However, when grouped with other buttons, it might make sense to style it to look a bit more like a button. Well, at least to have a border.

However, it's also possible that the links shouldn't be added to the group, but left separate. Here is an example of this:

![Buttons as two groups](https://user-images.githubusercontent.com/2327687/90601204-835cdd00-e200-11ea-903b-8f20cbfcc5b9.png)

**Comment by markkuriekkinen** [comment]
Now we actually have simple button groups (only for two buttons): #13 
They are not yet documented in the design system.
